pallets/external-validators/src/lib.rs

```diff
@@ -0,0 +1,697 @@
+            T::UpdateOrigin::ensure_origin(origin)?;
+            // don't let one unprepared collator ruin things for everyone.
+            let maybe_collator_id = T::ValidatorIdOf::convert(who.clone())
+                .filter(T::ValidatorRegistration::is_registered);
+
+            let collator_id = maybe_collator_id.ok_or(Error::<T>::NoKeysRegistered)?;
+
+            <WhitelistedValidators<T>>::try_mutate(|invulnerables| -> DispatchResult {
+                if invulnerables.contains(&collator_id) {
+                    Err(Error::<T>::AlreadyInvulnerable)?;
+                }
+                invulnerables
+                    .try_push(collator_id.clone())
+                    .map_err(|_| Error::<T>::TooManyInvulnerables)?;
+                Ok(())
+            })?;
+
+            Self::deposit_event(Event::WhitelistedValidatorAdded { account_id: who });
+
+            let weight_used = <T as Config>::WeightInfo::add_whitelisted(
+                WhitelistedValidators::<T>::decode_len()
+                    .unwrap_or_default()
+                    .try_into()
+                    .unwrap_or(T::MaxWhitelistedValidators::get().saturating_sub(1)),
+            );
+
+            Ok(Some(weight_used).into())
+        }
+
+        /// Remove an account `who` from the list of `WhitelistedValidators` collators.
+        ///
+        /// The origin for this call must be the `UpdateOrigin`.
+        #[pallet::call_index(2)]
+        #[pallet::weight(T::WeightInfo::remove_whitelisted(T::MaxWhitelistedValidators::get()))]
+        pub fn remove_whitelisted(origin: OriginFor<T>, who: T::AccountId) -> DispatchResult {
+            T::UpdateOrigin::ensure_origin(origin)?;
+
+            let collator_id = T::ValidatorIdOf::convert(who.clone())
+                .ok_or(Error::<T>::UnableToDeriveCollatorId)?;
+
+            <WhitelistedValidators<T>>::try_mutate(|invulnerables| -> DispatchResult {
+                let pos = invulnerables
+                    .iter()
+                    .position(|x| x == &collator_id)
+                    .ok_or(Error::<T>::NotInvulnerable)?;
+                invulnerables.remove(pos);
+                Ok(())
+            })?;
+
+            Self::deposit_event(Event::WhitelistedValidatorRemoved { account_id: who });
+            Ok(())
+        }
+
+        /// Force there to be no new eras indefinitely.
+        ///
+        /// The dispatch origin must be Root.
+        ///
+        /// # Warning
+        ///
+        /// The election process starts multiple blocks before the end of the era.
+        /// Thus the election process may be ongoing when this is called. In this case the
+        /// election will continue until the next era is triggered.
+        ///
+        /// ## Complexity
+        /// - No arguments.
+        /// - Weight: O(1)
+        #[pallet::call_index(3)]
+        #[pallet::weight(T::WeightInfo::force_no_eras())]
+        pub fn force_no_eras(origin: OriginFor<T>) -> DispatchResult {
```

Comment:
can we have a single extrinsic with these? (and the enum as an argument)

Reply:
*fixed*